framework/encode/vulkan_capture_manager.cpp

```diff
@@ -798,52 +799,56 @@ VkResult VulkanCaptureManager::OverrideCreateBuffer(VkDevice
-    VkBufferCreateFlags address_create_flags = 0;
-    VkBufferUsageFlags  address_usage_flags  = 0;
+    if (IsTrimEnabled())
+    {
+        modified_create_info->usage |= VK_BUFFER_USAGE_TRANSFER_SRC_BIT;
+
+        if (address_usage_flags2_KHR != nullptr)
+        {
+            *address_usage_flags2_KHR |= VK_BUFFER_USAGE_2_TRANSFER_SRC_BIT_KHR;
+        }
+    }
 
     if (device_wrapper->property_feature_info.feature_bufferDeviceAddressCaptureReplay)
     {
+        // If the buffer has shader device address usage, but the device address capture replay flag was not set, it
+        // needs to be set here.  We modify only a copy to prevent the modified pCreateInfo from being
+        // written to the capture file.
         if ((pCreateInfo->usage & VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT) ==
             VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT)
         {
             uses_address = true;
-            address_create_flags |= VK_BUFFER_CREATE_DEVICE_ADDRESS_CAPTURE_REPLAY_BIT;
+            modified_create_info->flags |= VK_BUFFER_CREATE_DEVICE_ADDRESS_CAPTURE_REPLAY_BIT;
         }
         if ((pCreateInfo->usage & VK_BUFFER_USAGE_ACCELERATION_STRUCTURE_STORAGE_BIT_KHR) ==
             VK_BUFFER_USAGE_ACCELERATION_STRUCTURE_STORAGE_BIT_KHR)
         {
             uses_address = true;
-            address_create_flags |= VK_BUFFER_CREATE_DEVICE_ADDRESS_CAPTURE_REPLAY_BIT;
-            address_usage_flags |= VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT;
-        }
-    }
+            modified_create_info->flags |= VK_BUFFER_CREATE_DEVICE_ADDRESS_CAPTURE_REPLAY_BIT;
+            modified_create_info->usage |= VK_BUFFER_USAGE_SHADER_DEVICE_ADDRESS_BIT;
 
-    // NOTE: VkBufferCreateInfo does not currently support pNext structures with handles, so does not have a handle
-    // unwrapping function.  If a pNext struct with handles is added in the future, it will be necessary to unwrap
-    // pCreateInfo before calling CreateBuffer.  The unwrapping process will create a mutable copy of the original
-    // pCreateInfo, with unwrapped handles, which can be modified directly and would not require the
-    // 'modified_create_info' copy performed below.
-    if (uses_address && (((pCreateInfo->flags & address_create_flags) != address_create_flags) ||
-                         ((pCreateInfo->usage & address_usage_flags) != address_usage_flags)))
-    {
-        // If the buffer has shader device address usage, but the device address capture replay flag was not set, it
-        // needs to be set here.  We create copy from an override to prevent the modified pCreateInfo from being
-        // written to the capture file.
-        modified_create_info.flags |= address_create_flags;
-        modified_create_info.usage |= address_usage_flags;
+            if (address_usage_flags2_KHR != nullptr)
+            {
+                *address_usage_flags2_KHR |= VK_BUFFER_USAGE_2_SHADER_DEVICE_ADDRESS_BIT_KHR;
```

Comment:
I don't think so. the original application used the pNext-chain to pass in [https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkBufferUsageFlags2CreateInfoKHR.html](VkBufferUsageFlags2CreateInfoKHR). this effectively overrides the old usage-flag, so we have to do the same in replay.  this option was added with `VK_KHR_maintenance5` but since we know the original application used it here, an additional check is not required for a faithful replay?

Reply:
Ah, good point!